packages/spacecat-shared-data-access/src/models/site/site.collection.js

```diff
@@ -71,6 +71,18 @@ class SiteCollection extends BaseCollection {
 
     return orderedSites;
   }
+
+  async findByPreviewURL(previewURL) {
+    const { hostname } = new URL(previewURL);
+    const [host] = hostname.split('.');
+    const [ref, site, owner] = host.split('--');
+
+    return this.findByHlxConfigRsoRefAndHlxConfigRsoSiteAndHlxConfigRsoOwner(
+      ref,
```

Comment:
@dzehnder , Do we need to worry about this part? This is the branch name of the related repo, correct?

Reply:
@dzehnder , What if a customer needs to validate changes against a non-`main` branch?
For example: `https://new-feature--da-bacom--adobecom.aem.page/drafts/narcis/dandd-london-case-study`